aws/resource_aws_servicecatalog_product.go

```diff
@@ -0,0 +1,447 @@
+	}
+
+	if output.ProductViewDetail == nil || output.ProductViewDetail.ProductViewSummary == nil {
+		return fmt.Errorf("error creating Service Catalog Product: no product view detail or summary")
+	}
+
+	if output.ProvisioningArtifactDetail == nil {
+		return fmt.Errorf("error creating Service Catalog Product: no provisioning artifact detail")
+	}
+
+	d.SetId(aws.StringValue(output.ProductViewDetail.ProductViewSummary.ProductId))
+
+	if _, err := waiter.ProductReady(conn, aws.StringValue(input.AcceptLanguage),
+		aws.StringValue(output.ProductViewDetail.ProductViewSummary.ProductId)); err != nil {
+		return fmt.Errorf("error waiting for product to be ready: %w", err)
+	}
+
+	return resourceAwsServiceCatalogProductRead(d, meta)
+}
+
+func resourceAwsServiceCatalogProductRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	output, err := waiter.ProductReady(conn, d.Get("accept_language").(string), d.Id())
+
+	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, servicecatalog.ErrCodeResourceNotFoundException) {
+		log.Printf("[WARN] Service Catalog Product (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error describing Service Catalog Product (%s): %w", d.Id(), err)
+	}
+
+	if output == nil || output.ProductViewDetail == nil || output.ProductViewDetail.ProductViewSummary == nil {
+		return fmt.Errorf("error getting Service Catalog Product (%s): empty response", d.Id())
+	}
+
+	pvs := output.ProductViewDetail.ProductViewSummary
+
+	d.Set("arn", output.ProductViewDetail.ProductARN)
+	if output.ProductViewDetail.CreatedTime != nil {
+		d.Set("created_time", output.ProductViewDetail.CreatedTime.Format(time.RFC3339))
+	}
+	d.Set("description", pvs.ShortDescription)
+	d.Set("distributor", pvs.Distributor)
+	d.Set("has_default_path", pvs.HasDefaultPath)
+	d.Set("name", pvs.Name)
+	d.Set("owner", pvs.Owner)
+	d.Set("status", output.ProductViewDetail.Status)
+	d.Set("support_description", pvs.SupportDescription)
+	d.Set("support_email", pvs.SupportEmail)
+	d.Set("support_url", pvs.SupportUrl)
+	d.Set("type", pvs.Type)
+
+	tags := keyvaluetags.ServicecatalogKeyValueTags(output.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig)
+
+	//lintignore:AWSR002
+	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %w", err)
+	}
+
+	if err := d.Set("tags_all", tags.Map()); err != nil {
+		return fmt.Errorf("error setting tags_all: %w", err)
+	}
+
+	return nil
+}
+
+func resourceAwsServiceCatalogProductUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+
+	if d.HasChanges(
+		"accept_language",
+		"description",
+		"distributor",
+		"name",
+		"owner",
+		"support_description",
+		"support_email",
+		"support_url",
+	) {
```

Comment:
More maintainable I think:

```suggestion
	if d.HasChangesExcept("tags", "tags_all") {
```